Chess/desk.h

```diff
@@ -1,25 +1,36 @@
 #ifndef DESK_H
 #define DESK_H
 
-#include <QObject>
+#include "common_types.h"
+#include "figure.h"
+
 #include <vector>
 
+
 class Desk : public QObject

```

Comment:
Why everything is Qobject? Can we have QT classes for GUI/Application skeleton only and not to use them inside model?

I admit I'm not a QT - developer. But my approach is to use specific library (whatever it is) only for dependent parts, and everything that can be independent - keep independent